### ISSUE TITLE
Unlike `pyflakes`, which correctly detects the problem, `redefined-while-unused (F811)` does not report duplicate imports when `if TYPE_CHECKING` is used

### ISSUE THREAD

**Issue by alessio-locatelli** [astral-sh/ruff#21888]
#### MRE

```py
from collections.abc import Iterator
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from collections.abc import Iterator


def func(_: Iterator[int]): ...
```

#### ruff behavior (it fails to detect the problem)

Command:

```bash
ruff check --select=F --preview mre.py
```

Output:
```
All checks passed!
```

Command:
```bash
ruff check --select=F mre.py
```

Output:
```
All checks passed!
```

#### `pyflakes` (it correctly detects the problem)

Command:
```bash
uvx pyflakes mre.py
```

Output:
```
mre.py:5:5: redefinition of unused 'Iterator' from line 1
```

### Version

0.14.8

**Comment by ntBre** [comment]
Thanks for the report and for the cool diagram! I think this is a duplicate of https://github.com/astral-sh/ruff/issues/17555.

**Comment by alessio-locatelli** [comment]
I apologize for creating a duplicate issue; I searched for duplicates using the issue title.

Searching the issue body at https://github.com/search?q=repo%3Aastral-sh%2Fruff%20F811%20is%3Aissue&type=issues yields 85 issues, which is simply impossible to review in an adequate amount of time.

@ntBre, if you have permission to edit the issue title, it would be greatly appreciated if you could include `F811` in the title of this issue: https://github.com/astral-sh/ruff/issues/17555. Perhaps this may help other people find an issue that is relevant to specific rule(s).

**Comment by ntBre** [comment]
No worries, the search can be a bit unreliable. I only knew because I'd seen the other issue before, and I guess searching for `f811 type_checking` got me there faster. I'll update the title.
